lib/modules/manager/gleam/artifacts.ts

```diff
@@ -49,7 +49,7 @@ export async function updateArtifacts(
       ],
     };
 
-    await exec('gleam deps download', execOptions);
+    await exec('gleam deps update', execOptions);
```

Comment:
does this update all dependencies to their latest allowed version?

Reply:
yes

Reply:
then this causes unwanted side effects for normal updates. update should only called when the update is a lockfile update.

@rarkins @secustor WDYT?

Reply:
The problem here is if a user has multiple dependencies which are unpinned in the lock file, and Renovate creates a PR which claims to update one of them, but actually updates all of them.

The gleam manager needs a way to specify one or more dependencies to update, e.g. `gleam deps update foo` to update package `foo`.

Reply:
Adding something like `gleam deps update <package name>` support to the Gleam CLI is probably the best permanent solution.

I don't yet know what is involved in adding this feature to Gleam. It would be great if there was some way to enable this functionality without waiting on a new Gleam feature.

In the past, have other managers worked around similar issues? Any alternative approaches I may not have thought about?

Reply:
Either:
1. Reverse engineer the lock file format (which is a bit risky), or
2. Document that it's not supported and recommend pinning dependencies instead

Reply:
This would unexpected behaviour by users and would look ( rightfully ) like a bug to them as `glean` would be the only package manager showing this behavior. 

I see these options to move forward: 
- parse manually the lock file ( as `@rarkins` mentioned  which is very risky and a lot of work to maintain )
- request a scoped update command for the Gleam CLI. This is for example how PDM lock files are updated. 
- Implement as is but make it opt in as an experimental flag. That way people are aware of the limitations before actually using.

Reply:
To reuse some Renovate terminology, if we merged this then it would _look like_ it's meant to update just one dependency, but actually each PR would be a "lock file maintenance". I think the confusion/misleadingness would be too much

Reply:
why not stay on `deps download ` like now when it's a normal update? so only necessary changes occur to the manifest

Reply:
Not a bad idea for the interim. I will do some testing and see if the behavior is acceptable.

I have also tagged this issue https://github.com/gleam-lang/gleam/issues/2170, and will work on getting this feature into gleam. That will provide the best permanent solution, but will take unknown time/effort.